presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcMetadata.java

```diff
@@ -200,7 +200,9 @@ public void rollback()
     @Override
     public ConnectorInsertTableHandle beginInsert(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
-        return jdbcClient.beginInsertTable(getTableMetadata(session, tableHandle));
+        JdbcOutputTableHandle handle = jdbcClient.beginInsertTable(getTableMetadata(session, tableHandle));
+        setRollback(() -> jdbcClient.rollbackCreateTable(handle));
```

Comment:
This needs to be a new `rollbackInsert` method. With this code, we would drop the table we are inserting into (not the temp table).

Reply:
Ah, you are right. I forgot that we actually use a temporary name and rename it at the end.

Reply:
Can we have  a test that ensures the target table remains alive?

Reply:
Tweaked the test I added so that it shows that.